onnxruntime/core/optimizer/embed_layer_norm_fusion.cc

```diff
@@ -160,51 +167,154 @@ Status EmbedLayerNormFusion::ApplyImpl(Graph& graph, bool& modified, int graph_l
+          {0, 0, "Unsqueeze", {1, 11}, kOnnxDomain},
+          {0, 0, "Cast", {9}, kOnnxDomain},
+          {0, 0, "Squeeze", {1}, kOnnxDomain},
+          {0, 0, "Transpose", {1}, kOnnxDomain},
+          {0, 0, "NonZero", {9}, kOnnxDomain},
+          {0, 0, "ConstantOfShape", {9}, kOnnxDomain},
+          {0, 0, "Unsqueeze", {1, 11}, kOnnxDomain},
+          {0, 0, "Gather", {1, 11}, kOnnxDomain},
+          {0, 0, "Shape", {1}, kOnnxDomain},
+      };
+      if (!graph_utils::FindPath(expand_node, true, pg_parent_path, pg_edges, logger)) {
+        continue;
+      }
+      for (size_t i = 0; i < pg_edges.size(); i++) {
+        if (pg_edges[i]->GetNode().GetOutputEdgesCount() != 1) {
+          isValidEmbedSubNode = false;
+          break;
+        }
+      }
+      // Check if the second input of the Gather node in the path has a constant input of 1
+      Node& gather_node = *graph.GetNode(pg_edges[pg_edges.size() - 2]->GetNode().Index());
+      if (!optimizer_utils::IsInitializerWithExpectedValue(graph, *(gather_node.InputDefs()[1]), int64_t(1), true)) {
+        DEBUG_LOG("Second input of Gather should be a constant with value 1. ");
+        
+        continue;
+      }
+      // Check if the two paths of position gather lead to the same input. 
+      Node& shape_node_2 = *graph.GetNode(pg_edges[pg_edges.size() - 1]->GetNode().Index());
+      if (shape_node_1.MutableInputDefs()[0] != shape_node_2.MutableInputDefs()[0]) {
+        continue;
       }
+      // Check if the parent of "shape" is the parent of "word gather"
+      if (shape_node_1.MutableInputDefs()[0] != word_gather_node.MutableInputDefs()[1]) {
+        continue;
+      }
+
+    }
+    if (!isValidEmbedSubNode) {
+      continue;
     }
 
     // Get input "input_ids" from node.
-    NodeArg* input_ids = CheckInput(graph, word_gather_node.MutableInputDefs()[1], layer_norm_node.GetExecutionProviderType(), logger);
-    if (input_ids == nullptr) {
+    NodeArg* input_ids = word_gather_node.MutableInputDefs()[1];
+    if (!CheckInput(input_ids, logger)) {
       DEBUG_LOG("Input id is not valid. ");
       continue;
     }
 
     // Get input "segment_ids" from node.
-    NodeArg* segment_ids = CheckInput(graph, segment_gather_node.MutableInputDefs()[1], layer_norm_node.GetExecutionProviderType(), logger);
-    if (segment_ids == nullptr) {
+    NodeArg* segment_ids = segment_gather_node.MutableInputDefs()[1];
+    if (!CheckInput(segment_ids, logger)) {
       DEBUG_LOG("Segment id is not valid. ");
       continue;
     }
 
     // Get input "mask" from "ReduceSum" node.
-    NodeArg* mask = CheckInput(graph, reduce_sum_node.MutableInputDefs()[0], layer_norm_node.GetExecutionProviderType(), logger);
-    if (mask == nullptr) {
+    NodeArg* mask = reduce_sum_node.MutableInputDefs()[0];
+    if (!CheckInput(mask, logger)) {
       DEBUG_LOG("Mask is not valid. ");
       continue;
     }
 
+    if (!input_ids->Shape()->dim()[1].has_dim_value()) {
```

Comment:
The constraint of dim 1 has been removed. Now, it can also be symbolic. So either it has same symbolic string or same dim value.

Could you try expression like input_ids->Shape() == segment_ids->Shape()?